syntax = "proto2";

package tf_trusted;

message GetModelLoadRequest {
  required string model_name = 1;
  required bytes model = 2;
}

message GetModelLoadResponse {}

enum ReturnType {
  NO_TYPE = 0;
  FLOAT = 1;
  DOUBLE = 2;
  INT32 = 3;
  INT64 = 4;
}


message GetModelPredictRequest {
  required ReturnType return_type = 1;
  required string model_name = 2;
  repeated float input = 3;
}

message GetModelPredictResponse {
  repeated float  float_result = 1;
  repeated double double_result = 2;
  repeated int32  int32_result = 3;
  repeated int64  int64_result = 4;

}

service Model {
  rpc GetModelLoad(GetModelLoadRequest) returns (GetModelLoadResponse) {}
  rpc GetModelPredict(GetModelPredictRequest) returns (GetModelPredictResponse) {}
}
